server.js

```diff
@@ -1,35 +1,87 @@
+
+import dotenv from 'dotenv'
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+dotenv.config()
+
+import books from './data/mybooks.json'
 
+//Setup to connect to our database
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+//Fields of our model
+const bookSchema = new mongoose.Schema ({
+  bookID: Number,
+  title: String,
+  authors: String,
+  num_pages: Number,
+  language: String,
+  sinopsis: String
+})
+
+//Model from booksSchema
+const Book = mongoose.model('Book', bookSchema)
+
+//Seeding of our database
+if (process.env.RESET_DB) {
+  
+  const seedDB = async () => {
+    await Book.deleteMany()
+
+    await books.forEach(item => {
+      const newBook = new Book(item)
+      newBook.save()
+    })
+  }    
+  seedDB()
+}
+  
 const port = process.env.PORT || 8080
-const app = express()
+const app = express() 
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hola world')
+})
+
+//Reaching for all the books, by title and author
+app.get('/books', async (req, res) => {
+  const { title, authors } = req.query
+ 
+  try { 
+    const books = await Book.find({ 
+      title: { 
+        $regex: new RegExp(title || "", "i")
+      }, 
+      authors: {
+        $regex: new RegExp(authors || "", "i")
+      }
+    })
+    res.json(books)    
+  }
+  catch (error) {
+        res.status(404).json({ error:'error'})
+      }
+})
+
+//Reaching one single book
+app.get('/books/:bookId', async (req, res) => {
+  const { bookId } = req.params
+
+  try {
+    const singleBook = await Book.findById(bookId)
```

Comment:
it is a little bit confusing because you have bookId in your model, but you using this name to filter by _id...